sinch/core/clients/sinch_client_configuration.py

```diff
@@ -53,6 +60,44 @@ def __init__(
         else:
             self.logger = logging.getLogger("Sinch")
 
+    def _set_sms_origin_with_service_plan_id(self):
+        self.sms_origin_with_service_plan_id = self._sms_domain_with_service_plan_id.format(self._sms_region)
+
+    def _set_sms_region_with_service_plan_id(self, region):
+        self._sms_region = region
+        self._set_sms_origin_with_service_plan_id()
+
+    def _get_sms_region_with_service_plan_id(self):
+        return self._sms_region
+
+    sms_region_with_service_plan_id = property(
+        _get_sms_region_with_service_plan_id,
+        _set_sms_region_with_service_plan_id,
+        doc="SMS Region for service plan id version of the SMS API"
+    )
+
+    def _set_sms_domain_with_service_plan_id(self, domain):
+        self._sms_domain = domain
+        self._set_sms_origin()
+
+    def _get_sms_domain_with_service_plan_id(self):
+        return self._sms_domain_with_service_plan_id
+
+    def set_sms_authentication_method(
```

Comment:
I understand that even if the customer is filling the servicePlanId and the API token, he will have to invoke this method on top. Would it be possible to determine which authentication method to use depending on the credentials / region ?

Reply:
I think we should discuss this API on our extended discussion meeting.
I'll add this subject to our agenda.

Reply:
It looks each of us have a different approach to that problem right now.
Until we figure out a unified solution I'm going to keep the code my way.

Reply:
Yes, this point is still in discussion: we can wait for a status before delivering a new feature.

Reply:
I've adapted the code to the draft ADR we have for that 👌